components/dashboard/src/Login.tsx

```diff
@@ -46,17 +47,33 @@ export function hasVisitedMarketingWebsiteBefore() {
 export function Login() {
     const { setUser } = useContext(UserContext);
     const { setTeams } = useContext(TeamsContext);
-    const showWelcome = !hasLoggedInBefore() && !hasVisitedMarketingWebsiteBefore();
+    const urlHash = getURLHash();
+    const repoPathname = urlHash ? new URL(urlHash).pathname : undefined;
+    let hostFromContext: string | undefined;
+    try {
+        hostFromContext = urlHash.length > 0 ? new URL(urlHash).host : undefined;
+    } catch (error) {
+        // Hash is not a valid URL
+    }
```

Comment:
The additional `new URL()` outside the try/catch can throw again. I'd suggest putting them in common inside the try/catch:

```suggestion
    let hostFromContext: string | undefined;
    let repoPathname: string | undefined;
    try {
        const url = new URL(urlHash);
        hostFromContext = url.host;
        repoPathname = url.pathname;
    } catch (error) {
        // Hash is not a valid URL
    }
```

Reply:
Of course! Thanks, this should've been obvious to me 🙈